eclipse-che/che-server
# feat: adding OAuth support to GitHub factory.

### What does this PR do?
This PR adds OAuth authentication support for GitHub factory.

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?
Issue [19966](https://github.com/eclipse/che/issues/19966)

### How to test this PR?

1. Make sure you have a linked GitHub account in Keycloak.
2. Create a private git repository on GitHub with a devfile in it
3. Try to open this private repository using the Che factory (whether by specifying the git repository or the path to the devfile)

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
